### PR TITLE
Pylons - Update configure pylons action

### PR DESCRIPTION
**Pull request by mharis001**
**When merged this pull request will:**
- Update configure pylons action to use `addActionToClass` instead of `addActionToObject` with `CBA_fnc_addClassEventHandler`
- Only use `createAction` once
- Update action condition to use `findIf` instead of `count` for checking rearm vehicles (should provide small performance benefit)

### CODE REVIEW

**Comment by mharis001** [comment]
@654wak654  Not sure if there was a special reason for using `addActionToObject` with init EH, but since it doesn't use inheritance it seems clearer to use `addActionToClass` and avoid the class EH.

File: addons/pylons/XEH_postInit.sqf [hunk 1]
```diff
@@ -2,28 +2,26 @@
 
 private _filter = "isClass (_x >> 'Components' >> 'TransportPylonsComponent') && {(getNumber (_x >> 'scope')) > 0}";
 GVAR(aircraftWithPylons) = (_filter configClasses (configFile >> "CfgVehicles")) apply {configName _x};
+
+GVAR(loadoutAction) = [
```
Comment (Dystopian) [inline addons/pylons/XEH_postInit.sqf]: why global not local?

Comment (mharis001) [reply addons/pylons/XEH_postInit.sqf]: oversight, fixed

**Comment by 654wak654** [comment]
@mharis001 Don't remember any special reason for using `addActionToObject`. It's probably just a decision I made while sleepy 😃.

**Comment by PabstMirror** [comment]
I don't really like how we're running `addActionToClass` on all possible classes (99% of witch won't exist in the mission).
With RHS+CUP `count ace_pylons_aircraftWithPylons = 316` and the section takes 200ms.

I think it would be better to just check aircraft on init and add it then
```
GVAR(aircraftWithPylons) = [];

GVAR(loadoutAction) = [
    QGVAR(loadoutAction),
    localize LSTRING(ConfigurePylons),
    "",
    {[_target] call FUNC(showDialog)},
    {
        if (!GVAR(enabledFromAmmoTrucks)) exitWith {false};

        private _vehicles = nearestObjects [_target, ["Air", "LandVehicle", "Slingload_base_F", "ReammoBox_F"], GVAR(searchDistance) + 10];
        private _filter = ["transportAmmo", QEGVAR(rearm,defaultSupply)] select (["ace_rearm"] call EFUNC(common,isModLoaded));
        private _cfgVehicles = configFile >> "CfgVehicles";
        private _rearmFound = _vehicles findIf {getNumber (_cfgVehicles >> typeOf _x >> _filter) > 0};

        _rearmFound > -1 && {[ACE_player, _target] call FUNC(canConfigurePylons)}
    }
] call EFUNC(interact_menu,createAction);

["Air", "init", {
    params ["_vehicle"];
    private _typeOf = typeOf _vehicle;

    if (_typeOf in GVAR(aircraftWithPylons)) exitWith {};
    if (!isClass (configFile >> "CfgVehicles" >> _typeOf >> 'Components' >> 'TransportPylonsComponent')) exitWith {};

    GVAR(aircraftWithPylons) pushBack _typeOf;
    [_typeOf, 0, ["ACE_MainActions"], GVAR(loadoutAction)] call EFUNC(interact_menu,addActionToClass);
}, true, ["ParachuteBase"], true] call CBA_fnc_addClassEventHandler;
```

**Comment by Dystopian** [comment]
@PabstMirror your code makes vehicle init a little slower. Not sure what is better: slow mission init or little slower vehicle init.

**Comment by dedmen** [comment]
^ Slow mission init. half a second longer loading screen is not a problem. But lags when placing vehicles are baaad.

**Comment by PabstMirror** [comment]
```
private _filter = "isClass (_x >> 'Components' >> 'TransportPylonsComponent') && {(getNumber (_x >> 'scope')) == 2}";
pTypes = (_filter configClasses (configFile >> "CfgVehicles")) apply {configName _x};
{
    _x createVehicle [0,50 * _forEachIndex, 0];
} forEach pTypes;
```
ace_pylons_init_counter: Average: 0.0168457s / 197 = 0.0855112ms
ace_pylons_init_counter: Average: 0.0128784s / 197 = 0.0653727ms

For duplicate vehicles, it would be even faster. IMHO, this isn't a problem

Compiling all actions for all vehicles not only takes time at start, but it also makes the interaction menu slower by adding cost to `getVariable` lookups if `count allVariables (ace_interact_menu_ActNamespace)` has hundreds of extra entries

**Comment by dedmen** [comment]
>slower by adding cost to `getVariable` lookups

Hashmap lookup is O(1) anyway.

File: addons/pylons/XEH_postInit.sqf [hunk 2]
```diff
@@ -2,28 +2,26 @@
 
 private _filter = "isClass (_x >> 'Components' >> 'TransportPylonsComponent') && {(getNumber (_x >> 'scope')) > 0}";
 GVAR(aircraftWithPylons) = (_filter configClasses (configFile >> "CfgVehicles")) apply {configName _x};
```
Comment (dedmen) [inline addons/pylons/XEH_postInit.sqf]: These two lines need 160ms to run (with my modset) 
Might be worth moving that to preStart and caching it.

**Comment by dedmen** [comment]
This stuff just showed up in profiler. 160ms for the config lookup and 370ms for the addClassEHs
Actually causes pylons postInit to be the worst of all,
tracy_2019-02-07_21-20-15

I'd say just implement @PabstMirror's idea.
We can later just profile what takes the most time at vehicle spawning and optimize that away, or revert this if it turns out so bad that it matters, though 0.1ms doesn't sound like it would matter much.

**Pull request merged**